app/assets/javascripts/single_page/dynamic_table.js.erb

```diff
@@ -192,44 +192,54 @@ const handleSelect = (e) => {
-            let splitted = [rowStatus.new, ""].concat(r.split(delimeter));
-            const missingColsCount = colsCount - defaultCols.length - splitted.length;
-            splitted = _defaultCols.concat(splitted);
-
-            // Preserve empty placeholder for columns of type of 'Registered Sample (multiple)'
-            sampleLinkIndexes.forEach((x) => splitted.splice(x, 0, []));
-
-            return missingColsCount < 0 ?
-              splitted.slice(0, colsCount) :
-              splitted.concat(Array(missingColsCount).fill(""));
-          })
-          .filter((x) => x);
-
-        this.table.rows.add(rows).draw();
+      try {
+        navigator.clipboard.readText().then((text) => {
+          const colsCount = this.table.columns().data().length;
+          const splitter = text.includes("\r\n") ? "\r\n" : "\n";
+          const delimeter = "\t";
+          const _defaultCols = defaultCols.map((c) => c.defaultValue || "");
+
+          const cols = this.table.settings()[0].aoColumns;
+          const sampleLinks = getSampleLinking(cols);
+          const sampleLinkIndexes = sampleLinks.map((x) => x.idx).sort((a, b) => b - a);
+          const sampleLinksTitles = sampleLinks.map((x) => x.title);
+
+          const rows = text
+            .split(splitter)
+            .filter((x) => x)
+            .map((r) => {
+              // Preserve row status, id and uuid
+              let splitted = [rowStatus.new, "", ""].concat(r.split(delimeter));
+              const missingColsCount = colsCount - defaultCols.length - splitted.length;
+              splitted = _defaultCols.concat(splitted);
+
+              // Preserve empty placeholder for columns of type of 'Registered Sample (multiple)'
+              sampleLinkIndexes.forEach((x) => splitted.splice(x, 0, []));
+
+              return missingColsCount < 0 ?
+                splitted.slice(0, colsCount) :
+                splitted.concat(Array(missingColsCount).fill(""));
+            })
+            .filter((x) => x);
+
+          this.table.rows.add(rows).draw();
+
+          if (sampleLinksTitles.length)
+            alert(
+              "No value is pasted for the following column(s): \n" +
+              sampleLinksTitles.map((x) => `"${x}"`).join(", ") +
+              "\nYou need to manually input them."
+            );
+        });
 
-        if (sampleLinksTitles.length)
-          alert(
-            "No value is pasted for the following column(s): \n" +
-            sampleLinksTitles.map((x) => `"${x}"`).join(", ") +
-            "\nYou need to manually input them."
-          );
-      });
+      } catch (error) {
+        if (error instanceof TypeError && navigator.userAgent.toLowerCase().includes('firefox')){
+          if(confirm('Your Firefox browser might not support clipboard access. By clicking OK, you\'ll be redirected to the official documentation.')){
+            window.open("https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API", "_blank");
```

Comment:
We have a HelpDictionary ( https://github.com/seek4science/seek/blob/main/lib/seek/help/help_dictionary.rb ) which helps us keep track of external links. A definition just needs adding to the https://github.com/seek4science/seek/blob/main/config/help_links.yml file and then look up by key with `Seek::Help::HelpDictionary.help_link(:key)`

Long term it helps keep track of external links, check them and update them when they inevitably change.

Reply:
Done! I replaced the URL with a help_link in the next commit.